media-api/app/lib/ImagePersistenceReasons.scala

```diff
@@ -100,17 +100,22 @@ object HasLeases extends PersistenceReason {
   override val reason: String = "leases"
 }
 
-case class IsInPersistedCollection(persistedCollections: List[String]) extends PersistenceReason {
-  override def shouldPersist(image: Image): Boolean = {
-    // list of the first element of each collection's `path`, i.e all the root collections
-    val collectionPaths: List[String] = image.collections.flatMap(_.path.headOption)
-
-    // is image in at least one persisted collection?
-    (collectionPaths diff persistedCollections).length < collectionPaths.length
+case class IsInPersistedCollection(maybePersistOnlyTheseCollections: Option[List[String]]) extends PersistenceReason {
+  override def shouldPersist(image: Image): Boolean = maybePersistOnlyTheseCollections match {
+    case None =>
+      image.collections.nonEmpty
+    case Some(persistedCollections) =>
+      // list of the first element of each collection's `path`, i.e all the root collections
+      val collectionPaths: List[String] = image.collections.flatMap(_.path.headOption)
+      // is image in at least one persisted collection?
+      (collectionPaths diff persistedCollections).length < collectionPaths.length
```

Comment:
looks like this config still only affects root collections? could we put a note to that effect somewhere, I'd say in our config doc but since it's patchy and not really looked at as is, maybe also/alternatively a comment by the code which parses the config key (ie. in CommonConfigWithElastic)

Reply:
great spot, I just tried seeing if we could actually match more than just the root collection, and it looks like the actual ES query is matching any part of the path AFAICT, so I'm tempted to adjust the reasons logic to better reflect the ES query, what do you think?

Reply:
oh great catch! Yes they should definitely match each other, either way (but I think your suggestion will probably be easier than making ES only query for the first item of the path)

Reply:
I did have a look to see if we could match full collection paths (e.g. `foo / bar / baz`) in the ES query but it looked hectic, so I've updated the reasons logic, should be building now...

Reply:
I've also added a comment to this effect in the config class